Includes/Core/Matrix/MatrixCSR-Impl.hpp

```diff
@@ -99,8 +99,32 @@ MatrixCSR<T>::MatrixCSR(const MatrixCSR& other)
 
 template <typename T>
 MatrixCSR<T>::MatrixCSR(MatrixCSR&& other) noexcept
+    : m_size(other.m_size),
+      m_nonZeros(std::move(other.m_nonZeros)),
+      m_rowPointers(std::move(other.m_rowPointers)),
+      m_columnIndices(std::move(other.m_columnIndices))
 {
-    (*this) = std::move(other);
+    other.m_size = Vector2UZ{};
+}
```

Comment:
Maybe it is trivial, How about use std::exchange instead of assign default value to other.m_size in constructor like below?
```c++
template <typename T>
MatrixCSR<T>::MatrixCSR(MatrixCSR&& other) noexcept
    : m_size(std::exchange(other.m_size, Vector2UZ{})),
      m_nonZeros(std::move(other.m_nonZeros)),
      m_rowPointers(std::move(other.m_rowPointers)),
      m_columnIndices(std::move(other.m_columnIndices))
{
    // Do nothing
}
```
With this modification, it is possible to suppress sonarcloud warning `Do not assign data members in a constructor. Initialize member "m_size" in an initialization list.` .

Reply:
Agree. I'll change it. 👍